QuizWhois.Domain/Services/Implementations/QuestionService.cs

```diff
@@ -30,42 +31,91 @@ public async Task<QuestionModel> AddQuestion(QuestionModel questionModel)
-            var entity = new Question { QuestionText = questionModel.QuestionText, CorrectAnswers = correctAnswers };
+            var entity = new Question { QuestionText = questionModel.QuestionText, CorrectAnswers = correctAnswers, PackId = questionModel.PackId };
             var result = await _context.Questions.AddAsync(entity);
             await _context.SaveChangesAsync();
             _logger.LogInformation($"Question id = {entity.Id} was added");
             var correctAnswersToModel = new List<string>();
             result.Entity.CorrectAnswers.ForEach(x => correctAnswersToModel.Add(x.AnswerText));
-            return new QuestionModel(result.Entity.Id, result.Entity.QuestionText, correctAnswersToModel);
+            return new QuestionModelResponse(result.Entity.Id, result.Entity.QuestionText, correctAnswersToModel, result.Entity.PackId);
         }
 
-        public QuestionModel GetQuestion(long questionId)
+        public QuestionModelResponse GetQuestion(long questionId)
         {
-            if (questionId <= 0)
-            {
-                throw new Exception("Id was invalid number");
-            }
+            DataValidation.ValidateId(questionId);
 
             var entity = _context.Questions.Where(x => x.Id == questionId).Select(x => new
                 {
                     id = x.Id,
                     questionText = x.QuestionText,
-                    correctAnswers = x.CorrectAnswers
+                    correctAnswers = x.CorrectAnswers,
+                    packId = x.PackId
                 }).FirstOrDefault();
 
             var correctAnswers = new List<string>();
             entity.correctAnswers.ToList().ForEach(x => correctAnswers.Add(x.AnswerText));
-            return new QuestionModel(entity.id, entity.questionText, correctAnswers);
+            return new QuestionModelResponse(entity.id, entity.questionText, correctAnswers, entity.packId);
+        }
+
+        public async Task UpdateQuestion(QuestionModelRequest questionModel, long questionId)
+        {
+            DataValidation.ValidateId(questionId);
+
+            var entity = _context.Questions.Where(x => x.Id == questionId).Select(x => new Question
+            {
+                Id = x.Id,
+                QuestionText = x.QuestionText,
+                CorrectAnswers = x.CorrectAnswers,
+                PackId = x.PackId
+            }).FirstOrDefault();
+            if (entity == null)
+            {
+                throw new Exception("Question is not found");
+            }
+
+            if (questionModel.QuestionText != string.Empty)
+            {
+                entity.QuestionText = questionModel.QuestionText;
+            }
+
+            if (questionModel.CorrectAnswers != null && questionModel.CorrectAnswers.Count != 0)
+            {
+                _context.Set<CorrectAnswer>().RemoveRange(entity.CorrectAnswers);
+                entity.CorrectAnswers.Clear();
```

Comment:
is it really necessary?

Reply:
Yes, otherwise i just add new Answers to old Answers, not rewrite them

Reply:
As long as the entity is tracked, you can call Clear and SaveChanges with no need to change CorrectAnswer set directly